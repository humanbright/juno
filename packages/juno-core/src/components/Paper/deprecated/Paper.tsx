import React, { FunctionComponent } from 'react';

import MuiPaper, { PaperProps as MuiPaperProps } from '@material-ui/core/Paper';

import { RcBaseProps } from '../../../foundation/typings';

export type RcPaperProps = RcBaseProps<MuiPaperProps>;

/** @deprecated please use release Paper */
export const RcPaper: FunctionComponent<RcPaperProps> & {
  dependencies?: any[];
} = React.memo((props) => <MuiPaper {...props} />);

RcPaper.displayName = 'RcPaper';
