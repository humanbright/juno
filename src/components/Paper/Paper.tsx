import MuiPaper from '@material-ui/core/Paper';
import React, { ComponentProps, forwardRef, useMemo } from 'react';

import {
  combineClasses,
  RcBaseProps,
  styled,
  useThemeProps,
} from '../../foundation';
import { PaperStyle } from './styles';
import { RcPaperClasses } from './utils';

type RcPaperProps = {} & RcBaseProps<ComponentProps<typeof MuiPaper>>;

const _RcPaper = forwardRef<any, RcPaperProps>((inProps, ref) => {
  const props = useThemeProps({ props: inProps, name: 'RcPaper' });
  const { classes: classesProp, children, ...rest } = props;
  const classes = useMemo(
    () => combineClasses(RcPaperClasses, classesProp),
    [classesProp],
  );

  return (
    <MuiPaper {...rest} ref={ref} classes={classes}>
      {children}
    </MuiPaper>
  );
});

const RcPaper = styled(_RcPaper)`
  ${PaperStyle}
`;

RcPaper.defaultProps = {};

RcPaper.displayName = 'RcPaper';

export { RcPaper };
export type { RcPaperProps };
