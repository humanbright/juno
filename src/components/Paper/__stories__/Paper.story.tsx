import React, { ComponentProps } from 'react';

import { Meta, Story } from '@storybook/react';

import {
  notControlInDocTable,
  notShowInDocTable,
  sortInDocTable,
  switchToControlKnobs,
} from '../../../storybook';
import { RcPaper } from '../Paper';

export default {
  title: 'ðŸš€ Cleanup Components/Paper',
  component: RcPaper,
  argTypes: {
    ...sortInDocTable<keyof PaperProps>([]),
    ...notControlInDocTable<keyof PaperProps>([]),
    ...notShowInDocTable<keyof PaperProps>([]),
  },
} as Meta;

type PaperProps = ComponentProps<typeof RcPaper>;

export const Paper: Story<PaperProps> = ({ children, ...args }) => {
  switchToControlKnobs();
  return (
    <RcPaper {...args}>
      <h3>This is a sheet of paper.</h3>
      <p>
        Paper can be used to build surface or other elements for your
        application.
      </p>
    </RcPaper>
  );
};

Paper.args = {
  elevation: 1,
};

Paper.argTypes = {
  ...notControlInDocTable<keyof PaperProps>([]),
};

Paper.parameters = {
  tags: [
    {
      name: 'Mui',
      href: 'https://material-ui.com/components/paper/#paper',
    },
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};
