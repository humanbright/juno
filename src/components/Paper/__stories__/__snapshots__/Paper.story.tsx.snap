// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Paper Paper 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <h3>
    This is a sheet of paper.
  </h3>
  <p>
    Paper can be used to build surface or other elements for your application.
  </p>
</div>
`;
